### PR TITLE
Progress towards first draft

### PR DESCRIPTION
**Pull request by drvinceknight**
There's a couple of things here but the main thing is a start on the Validation section: I've run a couple of simulations (using a small 200 repetitions cache) and shown that the theoretic fixation matches perfectly with the Moran one for some handpicked deterministic strategies. 

I've got a few stochastic ones (and have more running) that I'll be pushing in the morning (assuming they finish running).

All of this is is very much in draft form so I suggest that I keep going. My plan is to have a go at the introduction section tomorrow: you've made quite a few notes in there so that'll mainly help me come up to speed, although I note there you were still talking about the case of `i=1` and the differences that brings up so now that we're going to do `i=N/2` hopefully that'll clear that up :)

Anyway, I'll keep going and pushing to here. :+1:

### CODE REVIEW

**Comment by drvinceknight** [comment]
This branch now has quite a lot, and would be good to get your eyes on it when you have a moment @marcharper.

Still very early/drafty.

TLDR:

- Have an ok introduction in place (could do with more literature)
- Have written the validation part with some examples;
- Have written a script to summarise the data in to another data set and then written code in the nb to analyse (I didn't use your script for this, I felt it was straightforward using pandas but happy to change etc...)

I'm going to spend tomorrow (Sunday) working on this some more but we're getting to the point where we're either waiting on the data to finish running (all the plots/tables in the results section are made with incomplete runs of the data) or your input. :+1:

**Comment by drvinceknight** [comment]
I'm going to sit on this now until I hear from you Marc and/or get more data. :+1:

**Comment by marcharper** [comment]
This looks fine to me. I have a bunch of additions to make and suggestions for the plots. In genral I think we need to focus on a smaller number of high quality plots and to make sure we're sampling enough matches to eliminate statistical noise. Some of the new plots look a lot different than the former ones, the median ranks plot in particular -- the evolved strategies should have done better. Is this data from the approximate moran process?

Some major points to organize around:
* ZD strategies work well for very small populations but not in general
* Population size has a major impact
* Some strategies are good invaders / resistant to invaders across the board (the point of the boxplot diagram I had previously) -- we should find a good way to present this.

Feel free to merge. I'll make some edits soon.

**Comment by drvinceknight** [comment]
> This looks fine to me. I have a bunch of additions to make and suggestions for the plots. In genral I think we need to focus on a smaller number of high quality plots and to make sure we're sampling enough matches to eliminate statistical noise. Some of the new plots look a lot different than the former ones, the median ranks plot in particular -- the evolved strategies should have done better. Is this data from the approximate moran process?

Yeah I completely agree. I'm just throwing everything in right now.

Don't make any conclusions based on the graphs as they are just yet (that's why I haven't written anything). A lot of the strategies haven't played many matches for high N yet :)

Another reason the median rank plots might look different is that I'm breaking ties by using the average. Pandas has other options: http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.rank.html but I think this is a nice way of doing it.

> 
> Some major points to organize around:
> 
> ZD strategies work well for very small populations but not in general
> Population size has a major impact
> Some strategies are good invaders / resistant to invaders across the board (the point of the boxplot diagram I had previously) -- we should find a good way to present this.

Yeah I've been meaning to have a go at those. I might do that today.

> Feel free to merge. I'll make some edits soon.

Great :+1: Ok to continue to use PRs so that it's easy for us to keep track ? :) (I will)
